### PR TITLE
TSK-1083: updated HATEOAS from Version 0.24.0 to Version 1.0.4

### PR DESCRIPTION
**Pull request by krozzus**

### CODE REVIEW

File: history/taskana-simplehistory-rest-spring/pom.xml [hunk 1]
```diff
@@ -57,6 +57,16 @@
 			<artifactId>jackson-databind</artifactId>
 			<version>${version.jackson}</version>
 		</dependency>
+    <dependency>
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-rest-spring/pom.xml:60]: Why did you add these dependencies?

File: history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/rest/resource/TaskHistoryEventListAssembler.java [hunk 2]
```diff
@@ -18,27 +20,29 @@ public TaskHistoryEventListResource toResources(
       List<HistoryEventImpl> historyEvents, PageMetadata pageMetadata) {
 
     TaskHistoryEventAssembler assembler = new TaskHistoryEventAssembler();
-    List<TaskHistoryEventResource> resources = assembler.toResources(historyEvents);
+    List<TaskHistoryEventResource> resources =
+        new ArrayList<>(assembler.toCollectionModel(historyEvents).getContent());
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-rest-spring/src/main/java/pro/taskana/rest/resource/TaskHistoryEventListAssembler.java:24]: Why do you wrap the resources in an additional list?

File: pom.xml [hunk 3]
```diff
@@ -1,351 +1,353 @@
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (mustaphazorgati) [inline pom.xml:1]: Please reduce the diff to only the changes. (Do not format this file since the InteliJ formatter ruins the formatting)

Comment (mustaphazorgati) [reply pom.xml:1]: Tip: revert this file using `git checkout bb74e16187a5876ef46a91b00ae91304dda90b57 -- pom.xml` and then change the hate version

File: rest/taskana-rest-spring/pom.xml [hunk 4]
```diff
@@ -1,201 +1,207 @@
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/pom.xml:2]: Same as the other pom: reduce the diff to the changes only.
`git checkout bb74e16187a5876ef46a91b00ae91304dda90b57 -- rest/taskana-rest-spring/pom.xml`

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java [hunk 5]
```diff
@@ -196,13 +197,13 @@
...
       workbasket = workbasketService.updateWorkbasket(workbasket);
-      result = ResponseEntity.ok(workbasketResourceAssembler.toResource(workbasket));
+      result = ResponseEntity.ok(workbasketResourceAssembler.toModel(workbasket));
     } else {
       throw new InvalidWorkbasketException(
           "Target-WB-ID('"
               + workbasketId
               + "') is not identical with the WB-ID of to object which should be updated. ID=('"
-              + workbasketResource.getId()
+              + workbasketResource.getLink(IanaLinkRelations.SELF)
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java:206]: this should be workbasketResource.getWorkbasketId()

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/links/PageLinksAspect.java [hunk 6]
```diff
@@ -26,32 +27,35 @@
 @Aspect
 public class PageLinksAspect {
 
+  @SuppressWarnings("unchecked")
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/links/PageLinksAspect.java:30]: Do you want to try again and remove this supresswarning? Or should we leave it here for now?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/doc/api/AbstractPagingControllerRestDocumentation.java [hunk 7]
```diff
@@ -52,7 +52,7 @@ void setUp() {
               .description(pagingFieldDescriptionsMap.get("_links.first.href")),
           fieldWithPath("_links.last.href")
               .description(pagingFieldDescriptionsMap.get("_links.last.href")),
-          fieldWithPath("_links.prev.href")
+          fieldWithPath("_links.previous.href")
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/doc/api/AbstractPagingControllerRestDocumentation.java:55]: This should not be changed

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java (hunk 5 above)
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java:206]: From my understanding of the previous implementation of the function the getId() method came from the ResouceSupport class and returns a Type of Link with a relation of self. As such replacing it with a method that returns a type of String is not a valid replacement in my eyes.

**Comment by mustaphazorgati** [comment]
I found the issue for the current test failures: The new RepresentationModel now returns no page information. We used to have an adjustment within the old (copied) PagedResource:
``` @JsonProperty("page")
  public PageMetadata getMetadata() {
    if (Objects.isNull(metadata)) {
      Collection<T> content = getContent();
      return new PageMetadata(content.size(), 0, content.size());
    }
    return metadata;
  }

This will show always a page, even if we did not ask for one. - We should talk about @holgerhagen if we still want this behaviour or not.

**Comment by mustaphazorgati** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1083&id=krozzus_taskana
